jdbc/src/main/java/org/apache/zeppelin/jdbc/JDBCInterpreter.java

```diff
@@ -596,6 +598,24 @@ private String appendProxyUserToURL(String url, String user, String propertyKey)
     return connectionUrl.toString();
   }
 
+  // only add tags for hive jdbc
+  private String appendTagsToURL(String url, InterpreterContext context) {
+    StringBuilder builder = new StringBuilder(url);
+    if (url.startsWith("jdbc:hive2:")) {
+      Integer lastIndexOfQMark = builder.indexOf("?");
+      if (lastIndexOfQMark == -1) {
+        builder.append("?");
```

Comment:
There seems be no appending "?" mark if it doesn't exit at `appendProxyUserToURL()` line 585~586.

Reply:
Not sure what you mean. User may specify '?' in `default.url`